src/gui_common/charts/line/LineChart.cs

```diff
@@ -620,6 +633,14 @@ public void OverrideDataPointToolTipDescription(string dataset, DataPoint datapo
         }
     }
 
+    private void RemoveDataLineTooltipOnDeletion(DataLine line)
+    {
+        foreach (var tooltip in dataLineTooltips[line])
+        {
+            ToolTipManager.Instance.RemoveToolTip(tooltip.ToolTipNode.Name, LINE_CHART_GROUP_BASE_NAME);
+        }
+    }
```

Comment:
It's rarely that we need to remove a specific dataset and thus each line's tooltip separately so an all in one solution seems fitting.